### ISSUE TITLE
Cursor not a pointer when hovering over upload button

### ISSUE THREAD

**Issue by GoogleCodeExporter** [bdshadow/gwtupload#128]
```
In Firefox it would be nice if when hovering over the upload button the cursor 
was a pointer.

To reproduce problem:

1. Tested in Firefox 6 and 7
2. Using a MultiUploader with FileInputType.BUTTON
3. Hover mouse over button and cursor is a 'text' cursor, not a pointer (hand).

It works as expected in Chromium (version 14).

This non GWT ajax upload implementation has a pointer:

http://valums.com/ajax-upload/


Is it possible to resolve?
```

Original issue reported on code.google.com by `jasoncwi...@gmail.com` on 28 Oct 2011 at 5:02

**Comment by GoogleCodeExporter** [comment]
```
I was using GwtUpload 0.6.3
```

Original comment by `jasoncwi...@gmail.com` on 28 Oct 2011 at 6:04

**Comment by GoogleCodeExporter** [comment]
```
It is already a pointer cursor in the gwtupload implementation, download and 
run the examples, or test this url.

http://gwtupload.alcala.org/gupld/gwtuploadsample.SingleUploadSample/SingleUploa
dSample.html.
```

Original comment by `manuel.carrasco.m` on 9 Nov 2011 at 11:10
- Changed state: **Invalid**

**Comment by GoogleCodeExporter** [comment]
```
In Firefox 7.0.1, Ubuntu 11.10 on my computer the cursor is not a pointer. I've 
attached a photo of my screen.

My colleague checked on his computer (Firefox 7.0.1, Ubuntu 11.04) and the 
cursor for him was a pointer.

I'm not sure that the status of this issue should be set as invalid, as it is a 
real problem. It would be helpful at least to understand why my install of 
Firefox (and maybe others) is ignoring the CSS rule. If I find an answer I'll 
post back.

Thanks
```

Original comment by `jasoncwi...@gmail.com` on 9 Nov 2011 at 11:55

Attachments:
- [choose_a_file_cursor_ff7.JPG](https://storage.googleapis.com/google-code-attachments/gwtupload/issue-128/comment-3/choose_a_file_cursor_ff7.JPG)

**Comment by GoogleCodeExporter** [comment]
```
Sorry, I thought this issue was fixed, and seems a specific problem of certain 
FF version with certain version of OS.

This is not a matter of  css cursor rule because changing the css property will 
not change the cursor.

The problem in FF is a security issue which forces that the user must click on 
the '<file type=input>' html widget in order to open the choose window. 
What we do in gwtupload is to cover the real button with a giant file input 
with a opacity of 0, so when the user clicks on the button what he actually 
does is click on the file input.

The file input in FF has two parts the input for writing the name and the 
button, so we try by css move the input button over the decorated link so as 
the cursor is a pointer instead of a cursor. Note that neither cursor,  nor 
size can be changed by css, so I use fontsize and margin to play with sizes and 
positions.
In my linux machines and windowses FF cursor looks good, so proly your 
colleague has a different font configuration or whatever. I suggest you to open 
the firebug console and play with both the font-size and position of the input 
element until you see a pointer, then pls send me the optimum parameters and 
I'll change the library.

I leave the issue open until you give me an answer.



```

Original comment by `manuel.carrasco.m` on 9 Nov 2011 at 1:24
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Thanks, I'll play around with some parameters.
```

Original comment by `jasoncwi...@gmail.com` on 9 Nov 2011 at 9:26

**Comment by GoogleCodeExporter** [comment]
```
I was able to fix this by adding the following rule to the CSS:

input[type="file"] {
    cursor: pointer;
}


```

Original comment by `deborah0...@gmail.com` on 13 Mar 2012 at 5:06

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision r998.
```

Original comment by `manuel.carrasco.m` on 1 Jun 2012 at 10:19
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
This problem seems to have reappeared in Ubuntu 12.04 at least. Changing the 
font-size to 350 in Firebug changed the cursor back to a pointer for me. The 
previous fix is ineffective. My guess is that the system font changed from 
Ubuntu 10 to Ubuntu 11/12.
```

Original comment by `y2k...@gmail.com` on 18 Jul 2012 at 4:16

**Comment by GoogleCodeExporter** [comment]
Original comment by `manuel.carrasco.m` on 30 Nov 2012 at 9:15
- Added labels: **Milestone-0.6.5**

**Comment by GoogleCodeExporter** [comment]
```
I fixed it with the same rule in my CSS :

input[type="file"] {
    cursor: pointer;
}

thanks Deborah
```

Original comment by `kriss.ca...@dental-wings.com` on 5 Feb 2013 at 6:36

**Comment by GoogleCodeExporter** [comment]
```
input[type="file"] {
    cursor: pointer;
}

That won't fix it for me.
```

Original comment by `o...@sleepygiant.com` on 4 Apr 2013 at 10:57

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 1cb889395479.
```

Original comment by `manuel.carrasco.m` on 5 Apr 2013 at 9:49

**Comment by GoogleCodeExporter** [comment]
```
lolwat
```

Original comment by `pierceto...@gmail.com` on 11 Jan 2014 at 4:33
